rpc/src/main.rs

```diff
@@ -18,9 +18,9 @@ async fn main() -> Result<()> {
     let starknet_rpc = std::env::var("STARKNET_RPC_URL")
         .map_err(|_| eyre!("Missing mandatory environment variable: STARKNET_RPC_URL"))?;
 
-    let starknet_lightclient = StarknetClient::new(&starknet_rpc)?;
+    let starknet_lightclient = StarknetClientImpl::new(&starknet_rpc)?;
 
-    let (server_addr, server_handle) = run_server(starknet_lightclient).await?;
+    let (server_addr, server_handle) = run_server(Box::new(starknet_lightclient)).await?;
```

Comment:
why do we need to box it?

Reply:
Because of automock implementation. Bot exactly sure, but it requires it to be passed as a dyn and a good way was to use Box

Reply:
Ok!